### PR TITLE
bot-signup support

### PR DESCRIPTION
**Pull request by maxtaco**
- Bot signup doesn't make a device, it just makes a paper key
- Streamline signup to strip away flags and options no longer needed for bot mode
- slight hack: feed a paper key into DeviceWrap, but it seems to get the job done

### CODE REVIEW

**Comment by maxtaco** [comment]
Usage ideas:

```sh
keybase bot-signup -u tester44 > papereky
keybase logout
keybase oneshot -u tester44 < paperkey
keybase sign -m "it worked"
```

File: go/engine/signup.go [hunk 1]
```diff
@@ -46,6 +47,10 @@ type SignupEngineRunArg struct {
 	GenPGPBatch              bool // if true, generate and push a pgp key to the server (no interaction)
 	VerifyEmail              bool
 
+	// Bot signups have random PWs, no device keys, an eldest paper key, and return an paper key via
```
Comment (patrickxb) [inline go/engine/signup.go]: an paper -> a paper

File: go/engine/signup_test.go [hunk 2]
```diff
@@ -398,3 +400,68 @@ func TestSignupWithBadSecretStore(t *testing.T) {
...
+		Bot:                      true,
+	}
+
+	uis := libkb.UIs{
+		LogUI: tc.G.UI.GetLogUI(),
+	}
+	s := NewSignupEngine(tc.G, &arg)
+	m := NewMetaContextForTest(tc).WithUIs(uis)
+	err = RunEngine2(m, s)
+	require.NoError(tc.T, err)
```
Comment (patrickxb) [inline go/engine/signup_test.go]: Check that a paper key was generated?

Comment (maxtaco) [reply go/engine/signup_test.go]: Ok done!

**Pull request merged**